04_analysis.Rmd

```diff
@@ -384,12 +384,126 @@ This is a great format for manipulating/filtering and then graphing with ggplot
+
+### EXAMPLE :: Survey analysis 
+
+###### Step 0:: Call libraries
+
+``` {r, eval=T,tidy=FALSE, message= F, warning=F, error=F, echo=T}
+library(illuminate)
+library(tidyverse)
+library(purrr)
+library(readxl)
+library(openxlsx)
+library(srvyr)
+```
+
+
+
+
+###### Step 1:: Read data
+Read data with `read_sheets()`. This will make sure your data is stored as appropriate data type. It is important to make sure that all the select multiple are stored as logical, otherwise un weighted count will be wrong. It is recommended to use the `read_sheets()` to read the data. use ``?read_sheets()` for more details. 
+``` {r, tidy=FALSE, message= F, warning=F, error=F, echo=T}
+#  [recommended] read_sheets("[datapath]",data_type_fix = T,remove_all_NA_col = T,na_strings = c("NA",""," "))
+data_up<-read.csv("inputs/UKR2007_MSNA20_HH_dataset_main_rcop.csv")
+
+
+```
+The above code will give a dataframe  called `data_up`
+
+
+
+###### Step OPTIONAL:: Preaparing the data
+
+<span style="color: red;">ONLY APPLICABLE WHEN YOU ARE NOT READING YOUR DATA WITH `read_sheets()`</span>
+
+``` {r, eval=T,tidy=FALSE, message= F, warning=F, error=F, echo=T}
+# data_up <- read_excel("data/data.xlsx")
+data_up <- fix_data_type(df = data_up)
+```
+
+
+###### Step 2:: Weight calculation
+To do the weighted analysis, you will need to calculate the weights. If your dataset already have the weights column then you can ignore this part
+
+###### Read sampleframe
+``` {r, eval=F,tidy=FALSE, message= F, warning=F, error=F, echo=T}
+read_sheets("data/sampling_frame.xlsx")
+```
+
+
+This will give a dataframe called `sampling frame`
+
+``` {r, eval=F, tidy=FALSE, message= F, warning=F, error=F, echo=T}
+weights <- data_up %>% group_by(governorate1) %>% summarise(
+  survey_count = n()
+) %>% left_join(sampling_frame,by =c("governorate1"="strata.names")) %>% 
+  mutate(sample_global=sum(survey_count),
+         pop_global=sum(population),
+         survey_weight= (population/pop_global)/(survey_count/sample_global)) %>% select(governorate1,survey_weight)
+```
+
 
-## Analysis with numerical variables
-### Averages
-###Summarytools (CRAN package)
+
+###### Add weights to the dataframe
+``` {r, eval=FALSE, tidy=FALSE, message= F, warning=F, error=F, echo=T}
+data_up <- data_up %>% left_join(weights)
+```
+
+
+
+
+
+###### Step 3.1:: Weighted analysis
+
+``` {r, eval=FALSE, tidy=FALSE, message= F, warning=F, error=F, echo=T}
+overall_analysis <- survey_analysis(df = data_up,weights = T,weight_column = "survey_weight",strata = "governorate1")
+```
+
+###### Step 3.2:: Unweighted analysis
+
+``` {r, eval=T,tidy=FALSE, message= F, warning=F, error=F, echo=T}
+columns_to_analyze <- data_up[20:50] %>% names() 
+overall_analysis <- survey_analysis(df = data_up,weights = F,vars_to_analyze = columns_to_analyze )
+
```

Comment:
Also, maybe explicits the use of parameter "sm_sep" if you think it'd be useful for people?

 just did a small test on a specific dataset, if sep is "_", it could not find variables (I guess that's the desired behavior though!)